File: 50-powx-n/powx-n.java
class Solution {
    public double myPow(double x, int n) {
        if(x==0) return 0;
        long k = n<0 ? -1L * n : n;
        System.out.println(-1L * n);
        System.out.println(Math.abs(n));
        double ans = 1;
        while(k>0){
            if(k%2==0){
                x = x*x;
                k = k/2;
            }else{
                ans = ans*x;
                k--;
            }
        }

        return n<0 ? (1.0/ans) : ans;

    }
}

File: 50-powx-n/powx-n.py
class Solution(object):
    def myPow(self, x, n):
        def solve(x,n):
            if n == 0:
                return 1
            elif n < 0:
                return 1 / solve(x, -n)
        
            half = solve(x, n // 2)
            if n % 2 == 0:
                return half * half
            else:
                return x * half * half
            return solve(x,n)
        return solve(x,n)


        """
        :type x: float
        :type n: int
        :rtype: float
        """
        